### PR TITLE
Added PrivateAssets="all" to development dependencies

### PR DESCRIPTION
**Pull request by jp2masa**
## Changes
- Added PrivateAssets="all" to development dependencies.
- Updated RepoToolset.md.

### CODE REVIEW

File: docs/RepoToolset.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ Versions.props file is present in the repo and defines versions of all dependenc
     <UsingToolIbcOptimization>true</UsingToolIbcOptimization>
   
     <!-- RepoToolset package version -->
-    <RoslynToolsMicrosoftRepoToolsetVersion>1.0.0-alpha27</RoslynToolsMicrosoftRepoToolsetVersion>
+    <RoslynToolsRepoToolsetVersion>1.0.0-alpha27</RoslynToolsRepoToolsetVersion>
```
Comment (sharwell) [inline docs/RepoToolset.md]: :memo: The version itself seems to be old. Not sure if it needs to be updated now or later.

File: sdks/RepoToolset/tools/Compiler.props [hunk 2]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Net.Compilers" Version="$(MicrosoftNetCompilersVersion)" />
+    <PackageReference Include="Microsoft.Net.Compilers" Version="$(MicrosoftNetCompilersVersion)" PrivateAssets="all" />
```
Comment (jaredpar) [inline sdks/RepoToolset/tools/Compiler.props]: Can we elaborate on what was breaking here? I agree this is the right choice but also want to understand what impact this had (can't think of a prob it creates).

Comment (jp2masa) [reply sdks/RepoToolset/tools/Compiler.props]: `Microsoft.Net.Compilers` is a development dependency, but `PackageReference` doesn't have support for `developmentDependency` metadata yet. Currently, packages created from project files (like `Microsoft.VisualStudio.ProjectSystem.Managed`) pull `Microsoft.Net.Compilers`, but it shouldn't be included as a package dependency, as it's only used to build the projects.

**Comment by tmat** [comment]
👍 Thanks!

**Pull request merged**